packages/ra-ui-materialui/src/field/BooleanField.tsx

```diff
@@ -58,7 +58,7 @@ export const BooleanField: FC<BooleanFieldProps> = memo<BooleanFieldProps>(
                     {...sanitizeFieldRestProps(rest)}
                 >
                     <Tooltip title={translate(ariaLabel, { _: ariaLabel })}>
-                        {value === true ? (
+                        {value ? (
```

Comment:
no, it should be `value === true || (looseValue && value)`, otherwise, when `looseValue` is `false` and `value` is truthy but not true, it will show a TrueIcon, and that's a BC break.